components/suggestion/search_date_provider.tsx

```diff
@@ -4,15 +4,17 @@
 import Provider from './provider.jsx';
 import SearchDateSuggestion from './search_date_suggestion';
 
+type ResultsCallback = (results: {matchedPretext: string;terms: string[];items: Array<{label: string; date: string}>;component: typeof SearchDateSuggestion}) => void;
```

Comment:
Could we extract the items type here please?

```suggestion
type DateItem = {label: string; date: string};

type ResultsCallback = (results: {matchedPretext: string; terms: string[]; items: DateItem[]; component: typeof SearchDateSuggestion}) => void;
```